### PR TITLE
DC-243: Enable / disable cloud services independent of the Cloud Plugin itself

### PR DESCRIPTION
**Pull request by patrick-schweizer**
Jira: https://issues.opennms.org/browse/DC-243

### CODE REVIEW

File: plugin/src/main/java/org/opennms/plugins/cloud/tsaas/shell/EnableService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.plugins.cloud.tsaas.shell;
```
[the remark this answers is not in the document]
Comment (patrick-schweizer) [reply plugin/src/main/java/org/opennms/plugins/cloud/tsaas/shell/EnableService.java]: :+1: I restructured the packages

File: plugin/src/main/java/org/opennms/plugins/cloud/tsaas/shell/DisableService.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.plugins.cloud.tsaas.shell;
```
[the remark this answers is not in the document]
Comment (patrick-schweizer) [reply plugin/src/main/java/org/opennms/plugins/cloud/tsaas/shell/DisableService.java]: :+1: I restructured the packages

File: plugin/src/main/java/org/opennms/plugins/cloud/ServiceManager.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Responsible for registering cloud services at OpenNMS.
+ * Currently we support tsaas.
+ */
+public class ServiceManager {
```
[the remark this answers is not in the document]
Comment (patrick-schweizer) [reply plugin/src/main/java/org/opennms/plugins/cloud/ServiceManager.java]: done.

File: plugin/src/main/java/org/opennms/plugins/cloud/ServiceManager.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    private ServiceRegistration<TimeSeriesStorage> tsaasRegistration;
+
+    public ServiceManager(final BundleContext context,
+                          final TsaasStorage tsaas) {
+        this.context = Objects.requireNonNull(context);
+        this.tsaas = Objects.requireNonNull(tsaas);
+        registerTsaas(); // TODO: Patrick: this should be later conditional, depending if the client enabled tsaas
+    }
+
+    public void registerTsaas() {
```
[the remark this answers is not in the document]
Comment (patrick-schweizer) [reply plugin/src/main/java/org/opennms/plugins/cloud/ServiceManager.java]: fixed.

**Comment by patrick-schweizer** [comment]
@freddy61025,
thanks for the PR comments. I restructured the packages and made the service manager more generic.

**Review by freddy61025 — APPROVED** [review]
LGTM

**Pull request merged**